x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -364,7 +411,9 @@ export class AttachmentService {
            */
           references:
             refsWithExternalRefId.length > 0 || didDeleteOperation
-              ? refsWithExternalRefId
+              ? uniqBy([...refsWithExternalRefId, ...extractedReferences], 'id')
+              : extractedReferences.length > 0
+              ? extractedReferences
```

Comment:
I _think_ this could be simplified by moving the `uniqBy` call to line 399 and then doing:

```
references: refsWithExternalRefId.length > 0 || extractedReferences.length > 0 || didDeleteOperation ? uniquedRefs : undefined;
```

We might want to move the boolean computation or the whole construction of the refs to its own function though.